### DISCUSSION TITLE
Websockets help

### DISCUSSION THREAD

**Discussion by 0XPW87** [elysiajs/elysia#709] [Q&A]
Hi,

tried using \@teamwork/WebSocketJSONStream with the elysiajs .ws() path

`TypeError: ws.on is not a function. (In 'ws.on("message", function(msg) {`

Fair enough, probably caused by this not being sockets.io. Found this issue:  #162, however `io.attachApp(elysiaApp)` didn't work (port in use), and I can't figure out how to use the elysia path.

Clearly I am missing something. Should I use a wrapper? can I get Socket.io to work with Elysia?

Any help would be appreciated...

**Comment by crishoj** [comment]
Can you show how you use `.ws()` on the server?

**Reply by 0XPW87** [reply]
Hey,

it doesn’t really have anything to do with my specific usage, the question is if there is a wrapper or something that I can use to convert the elysia websocket to the socket.io interface, mostly for the `.on` listeners.

For this specific case I ended up writing my own websocketjson stream implementation, but it would be good to know for the future
